example-apps/middleware/index.js

```diff
@@ -15,14 +15,26 @@ const addSortingParams = (request /*, z */) => {
 };
 
 // HTTP after middleware that checks for errors in the response.
-const checkForErrors = (response, z) => {
-  // If we get a bad status code, throw an error. This will halt the zap.
-  if (response.status >= 300) {
-    throw new z.errors.HaltedError(
-      `Unexpected status code ${response.status} from ${response.request.url}`
+// When you don't use `afterResponse` it default to `[throwForStatus]`.
+const handleErrors = (response, z) => {
```

Comment:
Updated the middleware example to show:

1. Using `afterResponse` makes you responsible for error response handling.
2. How to throw for statuses that `throwForStatus` won't (correctly).
3. How to prevent `throwForStatus` from throwing.
4. That you should always call `throwForStatus`.